src/clients/kdump_proposal.rb

```diff
@@ -41,7 +41,24 @@ def main
 
       if @func == "MakeProposal"
         Kdump.Propose
-        @ret = { "raw_proposal" => Kdump.Summary }
+
+        @ret = {
+          "raw_proposal" => Kdump.Summary,
+          # FATE#317488 When expectation at the end of proposal does not match
+          # the value, this proposal will be called again
+          "trigger"      => {
+            "expect"     => "
+              Yast.import \"Kdump\"
+              Yast::Kdump.free_space_for_dump
+            ",
```

Comment:
with fun_ref it can look like

```
def expectation
  Yast.import "Kdump"
  Kdump.free_space_for_dump
end
...
...
"expect" => fun_ref(method(:expectation), "boolean ()")
```

this way it is much easier to test your expectation method and easier to read it.